File: leetcode/1108.cpp
// Runtime: 0 ms, faster than 100.00% of C++ online submissions for Defanging an IP Address.
// Memory Usage: 6.1 MB, less than 21.88% of C++ online submissions for Defanging an IP Address.
class Solution {
 public:
  string defangIPaddr(string address) {
    string res = "";
    for (string::iterator it = address.begin(); it != address.end(); ++it) {
      if (*it == '.') {
        res += "[.]";
      } else {
        res += *it;
      }
    }
    return res;
  }
};

File: leetcode/1108.py
"""
Runtime: 32 ms, faster than 46.86% of Python3 online submissions for Defanging an IP Address.
Memory Usage: 14.2 MB, less than 32.74% of Python3 online submissions for Defanging an IP Address.
"""


class Solution:
    def defangIPaddr(self, address: str) -> str:
        res = ""
        address = list(address)
        while address and address[0]:
            c = address.pop(0)
            res += c if c != "." else "[.]"

        return res
